exporter/otlphttpexporter/otlp.go

```diff
@@ -124,7 +125,9 @@ func (e *exporter) pushLogs(ctx context.Context, ld plog.Logs) error {
 }
 
 func (e *exporter) export(ctx context.Context, url string, request []byte) error {
-	e.logger.Debug("Preparing to make HTTP request", zap.String("url", url))
+	var span = trace.SpanFromContext(ctx)
+	var traceId = span.SpanContext().TraceID().String()
```

Comment:
Maybe we can do this function calls only when telemetry `service.telemetry.logs.level` is set to `DEBUG`?

Reply:
you can, use `zap.Stringer("traceId", traceId)` so delays the call to `String()`.

Reply:
Thank you @TylerHelmuth for being so supportive!   
Do you mean `func (ts *timeoutSender) send(req internal.Request) error` in exporthelper/common.go?
 
I see that is the common path for trace/log/metric export.

Reply:
@mcmho I don't think we have access to a logger from that function so there would need to be more work to add this feature to the helper.  @bogdandrutu what do you think; do we stick with otlphttpexporter (and otlpexporter in a future PR) for now, or try to provide this feature to all exporters from the get go?

Reply:
Thanks @TylerHelmuth!
@bogdandrutu ,  Is it not a good idea to keep a logger in 
```
// NewExporter creates a new Exporter.
func NewExporter(cfg ExporterSettings) *Exporter {
	return &Exporter{
		level:          cfg.ExporterCreateSettings.TelemetrySettings.MetricsLevel,
		spanNamePrefix: obsmetrics.ExporterPrefix + cfg.ExporterID.String(),
		mutators:       []tag.Mutator{tag.Upsert(obsmetrics.TagKeyExporter, cfg.ExporterID.String(), tag.WithTTL(tag.TTLNoPropagation))},
		tracer:         cfg.ExporterCreateSettings.TracerProvider.Tracer(cfg.ExporterID.String()),
	}
}
```

Reply:
Thanks @TylerHelmuth @bogdandrutu !

I appreciate your idea behind beauty of common code and reusability. 
For now, another need I see is that a logger would be useful for _otlpexporter_.oltp
so enabling it with a logger should be a good thing to do.

Currently both _otlphttpexporter_.exporter  and _loggingexporter_.loggingExporter have a logger,
```
type loggingExporter struct {
	logger     *zap.Logger
        ......  }
type exporter struct {
	logger     *zap.Logger
        ......  }
```
it might be good to make that symmetrical in otlpexporter.otlp.go as well:
type exporter struct {
	logger     *zap.Logger
        ......  }

And for longer term / future PR, we can attach to consolidate this feature to all exporters if we see fit.

How does that sound?   @bogdandrutu  @TylerHelmuth

Reply:
Wait... i just realized that `component.TelemetrySettings` inside `otlpexporter.exporter` has a logger.  
I can just use that with otlpexporter?

Reply:
Fixed by the latest commit.